### ISSUE TITLE
A person can be edited through `/api/v1/people`, if the correct body is passed

### ISSUE THREAD

**Issue by dianabarsan** [medic/cht-core#7780]
**Describe the bug**
[`/api/v1/people`](https://docs.communityhealthtoolkit.org/apps/reference/api/#post-apiv1people) is described as an endpoint that can create new people, and has an associated permission `can_create_people`. 
This endpoint _can_ be used to update an existing person if the correct body is sent, that contains the last `_rev` of the existent doc. Since there is no field validation, the endpoint can be used destructively, to delete an existent person document. 

Similarly, for places, we have two separate endpoints for create and edit, with the edit endpoint only allowing certain fields to be updated and using a different permission from the create endpoint.

**To Reproduce**
1. GET the contents of an existent person doc in the DB
2. Edit something, editing and adding new fields
3. POST this to `/api/v1/people`
4. Check the database and observe edits are saved

To delete, add a `_deleted: true` property to the document body.

**Expected behavior**
The endpoint should fail with some error if attempting to edit an existent document. 

**Environment**
- App: api
- Version: any?

**Comment by vikrantwiz02** [comment]
Hi @dianabarsan, @garethbowen, @andrablaj. 

I’d like to resolve this. I’ve identified that createPerson() in people.js lacks the _rev guard found in similar creation services. I plan to implement that check along with unit and integration tests to ensure a permanent fix. 

Could you assign this to me...?

**Comment by jkuester** [comment]
All of this sounds good.  👍   My only concern is if this should be considered a _breaking change_.  

@dianabarsan, maybe I am missing something, but I don't see any other endpoint for _updating a person_ (besides the brand new `PUT /api/v1/person/{id}` endpoint we just added.  I am worried that this "unintended" behavior might actually be the only way consumers could ever have updated persons via our REST api (meaning it might have more usage then we expect.... 🤔).  

If we decide this is a breaking change and we need to wait for `6.0.0` to ship it, then I think we should just totally remove the `/api/v1/people`.

**Comment by vikrantwiz02** [comment]
Agreed with your suggestion @jkuester! Since PUT /api/v1/person/:id already exists as the proper replacement, shipping this fix now to close the permission bypass and then fully removing POST /api/v1/people in 6.0.0 seems like the cleanest path. 

Let's see what @dianabarsan thinks!

**Comment by dianabarsan** [comment]
I agree with @jkuester 's caution. 
My argument against the breaking change is that we have not documented the endpoint as being an EDIT endpoint, just a create, and this issue is labeled as a bug. 

For safety, I'm fine with pushing it to 6.0.0 and removing the endpoint completely.

**Comment by vikrantwiz02** [comment]
Thanks @dianabarsan! I've updated the PR accordingly.  

- _rev guard approach has been dropped and POST /api/v1/people is now removed entirely.
- people import in routing.js is also cleaned up. createPerson() in people.js is left untouched since it's still used internally by getOrCreatePerson() (called by user-management and the create_user_for_contacts transition).
